app/src/main/java/ar/com/wolox/android/cookbook/notifications/helper/NotificationChannelFactory.kt

```diff
@@ -0,0 +1,48 @@
+package ar.com.wolox.android.cookbook.notifications.helper
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.os.Build
+import androidx.annotation.RequiresApi
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.wolmo.core.di.scopes.ApplicationScope
+import javax.inject.Inject
+
+@ApplicationScope
+@RequiresApi(Build.VERSION_CODES.O)
+class NotificationChannelFactory @Inject constructor() {
+
+    private val notificationChannels: MutableList<NotificationChannel> = arrayListOf()
+
+    fun createChannel(channelInfo: NotificationChannelInfo): NotificationChannel =
+        NotificationChannel(channelInfo.id, channelInfo.name, channelInfo.priority)
+            .apply {
+                this.description = channelInfo.description
+            }
+
+    fun init(context: Context) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
I think that since you have the `RequiresApi` above, you don't need this.

Reply:
The condition gonna be moved into the applicaction when the general channel is going to be created.